internal/compiler/generator/cpp.rs

```diff
@@ -3120,22 +3120,30 @@ fn compile_expression(expr: &llr::Expression, ctx: &EvaluationContext) -> String
             )
         }
         Expression::BinaryExpression { lhs, rhs, op } => {
-            let mut buffer = [0; 3];
-            format!(
-                "({lhs} {op} {rhs})",
-                lhs = compile_expression(lhs, ctx),
-                rhs = compile_expression(rhs, ctx),
-                op = match op {
-                    '=' => "==",
-                    '!' => "!=",
-                    '≤' => "<=",
-                    '≥' => ">=",
-                    '&' => "&&",
-                    '|' => "||",
-                    '/' => "/(float)",
-                    _ => op.encode_utf8(&mut buffer),
-                },
-            )
+            let lhs_str = compile_expression(lhs, ctx);
+            let rhs_str = compile_expression(rhs, ctx);
+
+            let lhs_ty = lhs.ty(ctx);
+
+            if (lhs_ty == Type::Float32 || lhs_ty.as_unit_product().is_some()) && (*op == '=' || *op == '!') {
```

Comment:
```suggestion
            if lhs_ty.as_unit_product().is_some() && (*op == '=' || *op == '!') {
```

Float32 is a unit product (an empty one, but an unit product nevertheless)

Reply:
Great :)